projects/vdk-plugins/vdk-notebook/src/vdk/plugin/notebook/notebook.py

```diff
@@ -46,28 +47,56 @@ class Notebook:
     Other cells are ignored.
     """
 
-    def __init__(self, file_path: Path):
+    @staticmethod
+    def register_notebook_steps(file_path: Path, context: JobContext):
         try:
-            self.sql_and_run_cells = []
-            self.python_helper_cells = []
-            self.file_path = file_path
+            python_module = imp.new_module("notebook")
+            exec("job_input = 1", python_module.__dict__)
+            notebook_steps = []
             # see Jupyter json schema here:
             # https://github.com/jupyter/nbformat/blob/main/nbformat/v4/nbformat.v4.schema.json
             content = json.loads(file_path.read_text())
+            index = 0
             for jupyter_cell in content["cells"]:
                 if jupyter_cell["cell_type"] == "code":
                     cell = Cell(jupyter_cell)
                     if CellUtils.is_vdk_cell(cell):
-                        if CellUtils.is_sql_cell(cell) or CellUtils.is_vdk_run_cell(
-                            cell
-                        ):
-                            self.sql_and_run_cells.append(cell)
+                        if CellUtils.is_sql_cell(cell):
+                            step = NotebookStep(
+                                name="".join(
+                                    [
+                                        file_path.name.replace(".ipynb", "_"),
+                                        str(index),
+                                    ]
+                                ),
+                                type=TYPE_SQL,
+                                runner_func=NotebookStepFuncFactory.run_sql_step,
+                                file_path=file_path,
+                                job_dir=context.job_directory,
+                                code=CellUtils.get_cell_code(cell),
+                            )
+                            notebook_steps.append(step)
+                            context.step_builder.add_step(step)
                         else:
-                            self.python_helper_cells.append(cell)
-            log.debug(
-                f"{len(self.sql_and_run_cells) + len(self.python_helper_cells)} "
-                f"cells with vdk tag were detected!"
-            )
+                            step = NotebookStep(
+                                name="".join(
```

Comment:
nit: step name seems the same in both if-else blocks; maybe calculate it in advance, then reuse it (deduplicate common code)